api/src/main/java/org/pmiops/workbench/api/ProfileController.java

```diff
@@ -449,13 +466,44 @@ private void verifyInvitationKey(String invitationKey){
     User user = userProvider.get();
     user.setGivenName(updatedProfile.getGivenName());
     user.setFamilyName(updatedProfile.getFamilyName());
+    user.setAboutYou(updatedProfile.getAboutYou());
+    user.setAreaOfResearch(updatedProfile.getAreaOfResearch());
+
     if (updatedProfile.getContactEmail() != null) {
       if (!updatedProfile.getContactEmail().equals(user.getContactEmail())) {
         mailChimpService.addUserContactEmail(updatedProfile.getContactEmail());
         user.setEmailVerificationStatus(EmailVerificationStatus.PENDING);
         user.setContactEmail(updatedProfile.getContactEmail());
       }
     }
+    List<org.pmiops.workbench.db.model.InstitutionalAffiliation> newAffiliations =
+        updatedProfile.getInstitutionalAffiliations()
+        .stream().map(FROM_CLIENT_INSTITUTIONAL_AFFILIATION)
+        .collect(Collectors.toList());
+    int i = 0;
+    ListIterator<org.pmiops.workbench.db.model.InstitutionalAffiliation> oldAffilations =
+        user.getInstitutionalAffiliations().listIterator();
+    boolean shouldAdd = false;
+    for (org.pmiops.workbench.db.model.InstitutionalAffiliation affiliation : newAffiliations) {
+      affiliation.setOrderIndex(i);
+      if (oldAffilations.hasNext()) {
+        org.pmiops.workbench.db.model.InstitutionalAffiliation oldAffilation = oldAffilations.next();
+        if (!oldAffilation.getRole().equals(affiliation.getRole())
+            || !oldAffilation.getInstitution().equals(affiliation.getInstitution())) {
+          shouldAdd = true;
```

Comment:
You can do break; here and in the else below.

Another way of doing this would be to make a Function<InstitutionalAffiliation, Pair<String, String>> and compare it applied to the elements of oldAffiliations vs. newAffiliations.

Reply:
I realized that I can't actually break here, because I am also doing other stuff in this loop. (Setting order, and also now setting userId)